### PR TITLE
[Localization] Modification some code about battle stat for localization

### PR DESCRIPTION
**Pull request by Ohmry**
## What are the changes?

Modification some code about battle stat for localization.

## Why am I doing these changes?

The message has been shown only english about battle stat such as ATK, DEF. so that it needs to modify that can be localized by translators I think.

## What did change?

- Modification return value of `getBattleStatName`, `getBattleStatLevelChangeDescription` method. it returns localized value used to i18next not English.
- Created `battle-stat.ts` in each locale folders for localizing by translators, they can translate messages about battle stat from this file.
- Created `battle-stat.test.ts` for test localization of battle stat.

#### Korean
스크린샷 2024-06-09 오후 1 46 59

#### Other languages
스크린샷 2024-06-09 오후 1 45 41

#### Result of test
image

## How to test the changes?

- Check out this PR and you can check when you use skill to change battle stat such as growl.
- In NPM, you can test using vitest, `npm run test battle-stat.test.ts`

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [ ] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Comment by CodeTappert** [comment]
The stat ones already exist in pokemon-info.ts 

Maybe use those instead? We dont need to have them localized twice

**Comment by Ohmry** [comment]
> The stat ones already exist in pokemon-info.ts
> 
> Maybe use those instead? We dont need to have them localized twice

You're right, I missed that. I modified localized key about battle stat that getting from pokemon-info not battle-stat. and I think battle-stat.ts is not necessary for storing one section. so that I moved level change description to battle.ts from battle-stat.ts. and I deleted all battle-stat.ts file too.

done test too.
image

**Comment by CodeTappert** [comment]
> > The stat ones already exist in pokemon-info.ts
> > Maybe use those instead? We dont need to have them localized twice
> 
> You're right, I missed that. I modified localized key about battle stat that getting from pokemon-info not battle-stat. and I think battle-stat.ts is not necessary for storing one section. so that I moved level change description to battle.ts from battle-stat.ts. and I deleted all battle-stat.ts file too.
> 
> done test too. image

You did not delete the german battle-stat.ts

**Comment by Ohmry** [comment]
> > > The stat ones already exist in pokemon-info.ts
> > > Maybe use those instead? We dont need to have them localized twice
> > 
> > 
> > You're right, I missed that. I modified localized key about battle stat that getting from pokemon-info not battle-stat. and I think battle-stat.ts is not necessary for storing one section. so that I moved level change description to battle.ts from battle-stat.ts. and I deleted all battle-stat.ts file too.
> > done test too. image
> 
> You did not delete the german battle-stat.ts

I deleted and push again, thanks.

**Comment by Greenlamp2** [comment]
Could you move your unit test into test/lokalisation please? thanks !

**Comment by Ohmry** [comment]
> Could you move your unit test into test/lokalisation please? thanks !

Sure, I did c8f8114

**Comment by CodeTappert** [comment]
> Could you move your unit test into test/lokalisation please? thanks !

Isnt the folder named wrong? Or is my english incorrect?

**Comment by Greenlamp2** [comment]
> > Could you move your unit test into test/lokalisation please? thanks !
> 
> Isnt the folder named wrong? Or is my english incorrect?

no, you're right, I'm used to using a service called "Lokalise" to manage translations, and the more I use it, the more it's changed in my mind how it's written ahah

**Review by returntoice — APPROVED** [review]
Okay enough for Korean translation.

File: src/data/battle-stat.ts [hunk 1]
```diff
@@ -34,30 +36,30 @@ export function getBattleStatLevelChangeDescription(levels: integer, up: boolean
   if (up) {
     switch (levels) {
     case 1:
-      return "rose";
+      return i18next.t("battle:statIsRose");
     case 2:
-      return "sharply rose";
+      return i18next.t("battle:statIsSharplyRose");
```
Comment (Tempo-anon) [inline src/data/battle-stat.ts:41]: nit: I'm not a fan of having `Is` in all of these. They can simply be `statRose` `statSharplyRose` `statRoseDrastically` `statWontGoAnyHigher` `statFell` `statHarshlyFell` `statSeverelyFell` and `statWontGoAnyLower`

Comment (Ohmry) [reply src/data/battle-stat.ts:41]: I will fix it and push again. and may I ask you check #1934? I merged commits that conflicted.

Comment (Ohmry) [reply src/data/battle-stat.ts]: I fixed. please check it.

**Review by Tempo-anon — DISMISSED** [review]

**Comment by Ohmry** [comment]
@Tempo-anon Hi, I resolved the conflicted. please confirm again.

**Review by Tempo-anon — APPROVED** [review]

**Review by Tempo-anon — APPROVED** [review]

**Pull request merged**